### PR TITLE
feat(Button): iTwinUI-CSS 1.0.0 refactor

### PR DESCRIPTION
**Pull request by gretanausedaite**
Adding changes made in [#684](https://github.com/iTwin/iTwinUI/pull/684)

### Button variants & types

- `<button class="iui-button iui-default">` -> `<button class="iui-button">`
- `<button class="iui-button iui-high-visibility">` -> `<button class="iui-button" data-iui-variant="high-visibility">`
- `<button class="iui-button iui-cta">` -> `<button class="iui-button" data-iui-variant="cta">`
- `<button class="iui-button iui-borderless">` -> `<button class="iui-button" data-iui-variant="borderless">`
- `<button class="iui-button iui-idea">` -> `<button class="iui-button" data-iui-variant="idea">`
- `<button class="iui-button iui-dropdown">` -> `<button class="iui-button iui-button-dropdown">`

### Button sizes

- `<button class="iui-button iui-small">` -> `<button class="iui-button" data-iui-size="small">`
- `<button class="iui-button iui-large">` -> `<button class="iui-button" data-iui-size="large">`

### Active button state

- `<button class="iui-button iui-active">` -> `<button class="iui-button" data-iui-active="true">`

### Disabled button states !! Not sure how to add this !!

```diff
<button disabled>
<button disabled="disabled">
+ <button aria-disabled="true">
```

### Button label

- `<span class="iui-button-label">Hello</span>` -> `<span>Hello</span>"`

### Split menu

- `<span class="iui-button-split-menu">` -> `<span class="iui-button-split">`

### CODE REVIEW

File: packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx [hunk 1]
```diff
@@ -71,14 +71,9 @@ export const Button: ButtonComponent = React.forwardRef((props, ref) => {
...
-        'iui-button',
-        `iui-${styleType}`,
-        {
-          [`iui-${size}`]: !!size,
-        },
-        className,
-      )}
+      className={cx('iui-button', className)}
+      data-iui-variant={styleType}
+      data-iui-size={size || undefined}
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx:76]: I think we don't want to add a data attribute for default variant and size, so we might need a ternary conditional here

File: packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx [hunk 2]
```diff
@@ -46,15 +46,10 @@ export const IconButton: IconButtonComponent = React.forwardRef(
...
-          `iui-${styleType}`,
-          {
-            [`iui-${size}`]: !!size,
-            'iui-active': isActive,
-          },
-          className,
-        )}
+        className={cx('iui-button', className)}
+        data-iui-variant={styleType}
+        data-iui-size={size || ''}
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx:51]: Same comment here

File: packages/iTwinUI-react/src/core/Alert/Alert.tsx [hunk 3]
```diff
@@ -96,14 +90,14 @@ export const Alert = (props: AlertProps) => {
...
-          size='small'
+        <button
           onClick={onClose}
           aria-label='Close'
+          type='button'
+          className='iui-alert-button'
         >
-          <SvgCloseSmall aria-hidden />
-        </IconButton>
+          <SvgCloseSmall aria-hidden className='iui-alert-button-icon' />
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/Alert/Alert.tsx:99]: Why did this file change? Shouldn't it be handled in #753?

Comment (gretanausedaite) [reply packages/iTwinUI-react/src/core/Alert/Alert.tsx:99]: It is handled there. I don't know why github sees this change. It does not show up in VS code PR 
image

File: packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx (hunk 1 above)
Comment (gretanausedaite) [reply packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx]: Size doesn't have `default` value. It's small or large. 
image

Added condition to variant data attribute.

File: packages/iTwinUI-react/src/core/Alert/Alert.tsx (hunk 3 above)
Comment (veekeys) [reply packages/iTwinUI-react/src/core/Alert/Alert.tsx:99]: I'd say branch is missing a merge with v1

Comment (gretanausedaite) [reply packages/iTwinUI-react/src/core/Alert/Alert.tsx]: Fixed

File: packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx [hunk 4]
```diff
@@ -71,14 +71,11 @@ export const Button: ButtonComponent = React.forwardRef((props, ref) => {
...
-        'iui-button',
-        `iui-${styleType}`,
-        {
-          [`iui-${size}`]: !!size,
-        },
-        className,
-      )}
+      className={cx('iui-button', className)}
+      data-iui-variant={
+        styleType && styleType !== 'default' ? styleType : undefined
```
Comment (veekeys) [inline packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx:76]: `styleType` is always defined, so enough
`styleType !== `default` ? styleType : undefined`

File: packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx [hunk 5]
```diff
@@ -46,15 +46,12 @@ export const IconButton: IconButtonComponent = React.forwardRef(
...
-            'iui-active': isActive,
-          },
-          className,
-        )}
+        className={cx('iui-button', className)}
+        data-iui-variant={
+          styleType && styleType !== 'default' ? styleType : undefined
+        }
+        data-iui-size={size}
+        data-iui-active={isActive}
```
Comment (veekeys) [inline packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx:54]: hmm... Interesting.. in comment, we said `isActive default value is false`, but we do not assign it in props destruction...
But it makes sense not to have attribute with `false` value, so lets leave it, I guess...

File: packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx (hunk 4 above)
Comment (gretanausedaite) [reply packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx]: updated

**Review by mayank99 — APPROVED** [review]

**Pull request merged**